data/Templates/Hl7v2/ADT_A01.liquid

```diff
@@ -2,14 +2,34 @@
     "resourceType": "Bundle",
     "type": "transaction",
     "entry": [
-        {% assign firstSegments = hl7v2Data | get_first_segments: 'PID|PD1|PV1|PV2|PR1|AVR' -%}
+        {% assign firstSegments = hl7v2Data | get_first_segments: 'PID|PD1|PV1|PV2|PR1|AVR|MSH' -%}
+
+        {% evaluate messageHeaderId using 'ID/MessageHeader' MSH: firstSegments.MSH -%}
+        {% if messageHeaderId -%}
+            {% include 'Resource/MessageHeader' MSH: firstSegments.MSH, ID: messageHeaderID -%}
+        {% endif -%}
+
+        {% evaluate provenanceId using 'ID/Provenance' MSH: firstSegments.MSH, baseID: patientId -%}
+        {% if provenanceId -%}
+            {% include 'Resource/Provenance' MSH: firstSegments.MSH, ORC: firstSegments.ORC, ID: provenanceId -%}
+        {% endif -%}
         
         {% evaluate patientId using 'ID/Patient' PID: firstSegments.PID, type: 'First' -%}
         {% if patientId -%}
             {% assign fullPatientId = patientId | prepend: 'Patient/' -%}
             {% include 'Resource/Patient' PID: firstSegments.PID, PD1: firstSegments.PD1, NK1: firstSegments.NK1, ID: patientId -%}
```

Comment:
It seems we are using NK1 (firstSegments.NK1) when generating Patient resource, but NK1 is not passed as a argument here in `get_first_segments`. Could you please help to double check about it?

Reply:
I think the NK1 segments(the first one and the others) wil be injected in the following line #67 (`{% for nk1Segment in nk1SegmentLists.NK1 -%}`), so we may not need to handle the NK1 separately here.

Reply:
Got it. If we are never using `firstSegments.NK1`, can we remove it from the arguments of Patient snippet?

Reply:
Sure.